proto/detect/configfork.go

```diff
@@ -0,0 +1,242 @@
+package detect
```

Comment:
At its core, this is SSZ detection it seems. Would prefer we have it under encoding/ssz/detect

Reply:
I see what you mean and tend to agree, but because this package needs to initialize protos, conceptually does moving it over into encoding/ssz get a bit circular, considering the state / proto packages need to import from ssz?

Reply:
Actually I think I agree with you, I'll try moving this around to see how it feels. Thanks for that suggestion.

Reply:
forgot to update, i made this move :+1: